components/eamxx/src/share/io/scorpio_input.cpp

```diff
@@ -350,7 +350,13 @@ void AtmosphereInput::finalize()
 /* ---------------------------------------------------------- */
 void AtmosphereInput::init_scorpio_structures() 
 {
-  scorpio::register_file(m_filename,scorpio::Read);
+  int iotype = scorpio::str2iotype("default");
+  if(m_params.isParameter("iotype")){
```

Comment:
If you want, you can be more concise:
```suggestion
auto iotype_str = m_params.get<std::string>("iotype","default");
int iotype = scorpio::str2iotype(iotype_str);
```

Reply:
Thanks, your suggestion looks better/concise. I will make that change.